sound/soc/sof/sof-pci-dev.c

```diff
@@ -225,7 +225,11 @@ EXPORT_SYMBOL_NS(sof_pci_remove, SND_SOC_SOF_PCI_DEV);
 
 void sof_pci_shutdown(struct pci_dev *pci)
 {
-	snd_sof_device_shutdown(&pci->dev);
+	struct snd_sof_dev *sdev = dev_get_drvdata(&pci->dev);
+
+	/* audio need to enter D3 before shutdown */
+	sdev->system_suspend_target = SOF_SUSPEND_S3;
+	snd_sof_suspend(&pci->dev);
```

Comment:
NAK, we added a platform-specific callback that you removed, so you just killed what @keyonjie added in January.
You guys are in the same timezone, can you please talk with each other?

Reply:
> NAK, we added a platform-specific callback that you removed, so you just killed what @keyonjie added in January.
> You guys are in the same timezone, can you please talk with each other?

Yes, @libinyang if we decide to change this to a generic solution for all platforms, we need to remove the platform specific ones, let's have a talk about this.

Reply:
> > NAK, we added a platform-specific callback that you removed, so you just killed what @keyonjie added in January.
> > You guys are in the same timezone, can you please talk with each other?
> 
> Yes, @libinyang if we decide to change this to a generic solution for all platforms, we need to remove the platform specific ones, let's have a talk about this.

@plbossart @keyonjie OK. I will talk with @keyonjie what to make the patch more reasonable and acceptable.